llmfoundry/data/finetuning/dataloader.py

```diff
@@ -145,7 +148,34 @@ def build_finetuning_dataloader(cfg: DictConfig, tokenizer: Tokenizer,
         )
 
     else:
-        dataset = dataset_constructor.build_from_hf(cfg.dataset, tokenizer)
+        backend, _, _ = parse_uri(cfg.dataset.hf_name)
+        if backend not in ['', None]:
+            if cfg.dataset.get('split') is None:
+                raise ValueError(
+                    'When using a HuggingFace dataset from a URL, you must set the ' +\
+                    '`split` key in the dataset config.'
+                )
+            with tempfile.TemporaryDirectory() as tmp_dir:
+                supported_extensions = ['jsonl', 'csv', 'parquet']
+                for extension in supported_extensions:
+                    name = f'{cfg.dataset.hf_name}/{cfg.dataset.split}.{extension}'
+                    destination = f'{tmp_dir}/{name}'
+                    try:
+                        with dist.local_rank_zero_download_and_wait(
+                                destination):
+                            if dist.get_local_rank() == 0:
+                                get_file(name, destination, overwrite=True)
+                    except FileNotFoundError as e:
+                        if extension == supported_extensions[-1]:
+                            raise e
```

Comment:
can we raise a nicer error here? otherwise the last filenotfound will be about a parquet path, which might be confusing to a user if they were (e.g.) trying to use a jsonl.